modules/cnf-image-based-upgrade.adoc

```diff
@@ -102,48 +107,50 @@ During the `Prep` stage, you specify the following upgrade details in the `Image
 
-[IMPORTANT]
-====
-The same version of the applications must function on both the current and the target release of {product-title}.
-====
+Additionally to the OADP Operator, the {lcao} uses the `ostree` versioning system to create a backup, which allows complete cluster reconfiguration after both upgrade and rollback.
 
-Additionally to the OADP Operator, the {lcao} uses the `ostree` versioning system to create a backup, which allow complete cluster reconfiguration after both upgrade and rollback.
-
-You can stop the upgrade process at this point by moving to the `Idle` stage or you can start the upgrade by moving to the `Upgrade` stage in the `ImageBasedUpgrade` CR .
+You can cancel the upgrade process at this point by moving to the `Idle` stage or you can start the upgrade by moving to the `Upgrade` stage in the `ImageBasedUpgrade` CR.
 If you stop, the Operator performs cleanup operations.
 
+image::../images/ibu-prep.png[Transition from Prep stage]
+
 [id="ztp-image-based-upgrade-concept-upgrade_{context}"]
-== Upgrade stage
+=== Upgrade stage
 
-Just before the {lcao} starts the upgrade process, the backup of your cluster resources specified in the `Prep` stage are created on a compatible Object storage solution.
-After the target cluster reboots with the new platform version, the Operator applies the cluster and application configurations defined in the `Backup` and `Restore` CRs, and applies any extra manifests that are specified in the referenced `ConfigMap` resource.
+The `Upgrade` stage consists of three phases:
 
+pre-pivot:: Just before pivoting to the new stateroot, the {lcao} collects the required cluster specific artifacts and stores them in the new stateroot. The backup of your cluster resources specified in the `Prep` stage are created on a compatible Object storage solution. The {lcao} exports CRs specified in the `extraManifests` field in the `ImageBasedUpgrade` CR or the CRs described in the ZTP policies that are bound to the target cluster.
+pivot:: Once pre-pivot phase is completed, the {lcao} makes the new state root as default and reboots the node.
+post-pivot:: After booting from the new stateroot, the {lcao} reconfigures the cluster by applying cluster-specific artifacts that were collected in the pre-pivot phase. The Operator applies all saved CRs, and restores the backups.
 The Operator also regenerates the seed image's cluster cryptography.
 This ensures that each {sno} cluster upgraded with the same seed image has unique and valid cryptographic objects.
 
 Once you are satisfied with the changes, you can finalize the upgrade by moving to the `Idle` stage.
+
+image::../images/ibu-finalize-upgrade.png[Finalize upgrade]
+
+If you want to cancel the upgrade, you can do so until the pre-pivot phase of the `Upgrade` stage.
 If you encounter issues after the upgrade, you can move to the `Rollback` stage for a manual rollback.
 
+image::../images/ibu-upgrade-rollback-cancel.png[Rollback upgrade]
```

Comment:
There are too many images in this module. If possible, condense what you are tying to convey into a single image.